sdk/identity/src/token_credentials/azureauth_cli_credentials.rs

```diff
@@ -0,0 +1,185 @@
+use azure_core::{
+    auth::{AccessToken, TokenCredential, TokenResponse},
+    error::{Error, ErrorKind, ResultExt},
+};
+use oauth2::ClientId;
+use serde::Deserialize;
+use std::process::Command;
+use std::str;
+use time::OffsetDateTime;
+
+mod unix_date_string {
+    use azure_core::error::{Error, ErrorKind};
+    use serde::{Deserialize, Deserializer};
+    use time::OffsetDateTime;
+
+    pub fn parse(s: &str) -> azure_core::Result<OffsetDateTime> {
+        let as_i64 = s.parse().map_err(|_| {
+            Error::with_message(ErrorKind::DataConversion, || {
+                format!("unable to parse expiration_date '{s}")
+            })
+        })?;
+
+        OffsetDateTime::from_unix_timestamp(as_i64).map_err(|_| {
+            Error::with_message(ErrorKind::DataConversion, || {
+                format!("unable to parse expiration_date '{s}")
+            })
+        })
+    }
+
+    pub fn deserialize<'de, D>(deserializer: D) -> Result<OffsetDateTime, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        let s = String::deserialize(deserializer)?;
+        parse(&s).map_err(serde::de::Error::custom)
+    }
+}
+
+#[derive(Debug, Clone, Deserialize)]
+struct CliTokenResponse {
+    pub token: AccessToken,
+    #[serde(with = "unix_date_string")]
+    pub expiration_date: OffsetDateTime,
+}
+
+/// Authentication Mode
+///
+/// Note: While the azureauth CLI supports devicecode, users wishing to use
+/// devicecode should use `azure_identity::device_code_flow`
+#[derive(Debug, Clone, Copy)]
+pub enum AzureauthCliMode {
+    All,
+    #[cfg(target_os = "windows")]
+    IntegratedWindowsAuth,
+    Broker,
+    Web,
+}
+
+/// Enables authentication to Azure Active Directory using Azure CLI to obtain an access token.
+pub struct AzureauthCliCredential {
+    tenant_id: String,
+    client_id: ClientId,
+    modes: Vec<AzureauthCliMode>,
+}
+
+impl AzureauthCliCredential {
+    /// Create a new `AzureCliCredential`
+    pub fn new<T, C>(tenant_id: T, client_id: C) -> Self
+    where
+        T: Into<String>,
+        C: Into<ClientId>,
+    {
+        Self {
+            tenant_id: tenant_id.into(),
+            client_id: client_id.into(),
+            modes: Vec::new(),
+        }
+    }
+
+    pub fn add_mode(mut self, mode: AzureauthCliMode) -> Self {
+        self.modes.push(mode);
+        self
+    }
+
+    pub fn with_modes(mut self, modes: Vec<AzureauthCliMode>) -> Self {
+        self.modes = modes;
+        self
+    }
+
+    fn get_access_token(&self, resource: &str) -> azure_core::Result<CliTokenResponse> {
```

Comment:
Could we take `scopes: &Vec<String>` or similar here instead of a single `resource`?